functests/dpdk/dpdk.go

```diff
@@ -201,15 +294,363 @@ func getNumberOfPackets(line, firstFieldSubstr string) int {
 	return d
 }
 
-// findDPDKWorkloadPod finds a pod running a DPDK application
-func findDPDKWorkloadPod() *corev1.Pod {
-	listOptions := metav1.ListOptions{
-		LabelSelector: labels.SelectorFromSet(labels.Set{"deploymentconfig": "s2i-dpdk-app"}).String(),
+func GetOrCreateDPDKWorkload() *corev1.Pod {
```

Comment:
This does a lot more than it's name says.
Can you split it and call the different functions?

Like 
- tryToFindDPDKPod
- findOrOverridePerformanceProfile
- findOrOverrideSriovNetwork
- createPod